File: scripts/export-db-to-json.js
const Database = require('better-sqlite3');
const fs = require('fs');
const path = require('path');

const dbPath = path.resolve(__dirname, '../server/db/repositories.db');
const outputPath = path.resolve(__dirname, '../public/repositories_details.json');

try {
    const db = new Database(dbPath, { readonly: true });
    const repos = db.prepare('SELECT * FROM repositories').all();

    const formatted = repos.reduce((acc, repo) => {
        acc[repo.id] = {
            ...repo,
            topics: repo.topics ? repo.topics.split('|') : [],
            languages: repo.languages ? repo.languages.split('|') : []
        };
        return acc;
    }, {});

    fs.writeFileSync(outputPath, JSON.stringify(formatted, null, 2));
    console.log(`✅ Exported ${repos.length} repositories to ${outputPath}`);
    console.log(`📦 File size: ${(fs.statSync(outputPath).size / 1024).toFixed(2)} KB`);

    db.close();
} catch (error) {
    console.error('❌ Error exporting database:', error);
    process.exit(1);
}

File: scripts/export-db-to-json.py
import sqlite3
import json
import os

db_path = os.path.join(os.path.dirname(__file__), '../server/db/repositories.db')
output_path = os.path.join(os.path.dirname(__file__), '../public/repositories_details.json')

try:
    conn = sqlite3.connect(db_path)
    conn.row_factory = sqlite3.Row
    cursor = conn.cursor()
    
    cursor.execute("SELECT * FROM repositories")
    rows = cursor.fetchall()
    
    formatted = {}
    for row in rows:
        repo_dict = dict(row)
        repo_id = repo_dict['id']
        
        # Convert pipe-separated strings to arrays
        repo_dict['topics'] = repo_dict['topics'].split('|') if repo_dict.get('topics') else []
        repo_dict['languages'] = repo_dict['languages'].split('|') if repo_dict.get('languages') else []
        
        formatted[repo_id] = repo_dict
    
    with open(output_path, 'w') as f:
        json.dump(formatted, f, indent=2)
    
    file_size = os.path.getsize(output_path) / 1024
    print(f"✅ Exported {len(rows)} repositories to {output_path}")
    print(f"📦 File size: {file_size:.2f} KB")
    
    conn.close()
except Exception as error:
    print(f"❌ Error exporting database: {error}")
    exit(1)
